packages/extension-browser/src/devtools/views/controls/settings.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { FormEvent, useCallback } from 'react';
+
+import { getMessage } from '../../utils/i18n';
+
+import Summary from './summary';
+import ConfigLabel from '../pages/config/label';
+import Checkbox from './checkbox';
+import LabelText from './label-text';
+
+import * as styles from './settings.css';
+
+const settings = { telemetry: 'telemetry' };
+
+type Props = {
+    isTelemetryEnabled: boolean;
+    onTelemetryChange: (enable?: boolean) => void;
+};
+
+const Settings = ({isTelemetryEnabled, onTelemetryChange}: Props) => {
+
+    const onEnabledChange = useCallback((event: FormEvent<HTMLInputElement>) => {
+        const input = (event.target as HTMLInputElement);
+        const setting = input.value;
+
+        if (setting === settings.telemetry) {
```

Comment:
This check seems unnecessary. You're only assigning this handler to a checkbox which is intended for telemetry, so you shouldn't need to provide or check `input.value`.

Reply:
This was thinking in the future with more settings, but I can remove it if you want to.

Reply:
Yes, I would expect each setting to have its own callback so let's remove it.

Reply:
Then this shouldn't be here. I should create a new component for this setting in particular.